be/src/vec/exec/format/parquet/delta_bit_pack_decoder.cpp

```diff
@@ -0,0 +1,266 @@
+        // TODO: The Slice to be decoded will not reuse so don't need skip the padding bits,
+        // but we can also skip them for the robustness.
+        _values_remaining_current_mini_block = 0;
+    }
+    *out_num_values = num_values;
+    return Status::OK();
+}
+
+void DeltaLengthByteArrayDecoder::_decode_lengths() {
+    _len_decoder.set_bit_reader(_bit_reader);
+    // get the number of encoded lengths
+    int num_length = _len_decoder.valid_values_count();
+    _buffered_length.resize(num_length * sizeof(int32_t));
+
+    // decode all the lengths. all the lengths are buffered in buffered_length_.
+    int ret;
+    _len_decoder.decode(reinterpret_cast<int32_t*>(_buffered_length.data()), num_length, &ret);
+    DCHECK_EQ(ret, num_length);
+    _length_idx = 0;
+    _num_valid_values = num_length;
+}
+
+Status DeltaLengthByteArrayDecoder::_get_internal(Slice* buffer, int max_values,
+                                                  int* out_num_values) {
+    // Decode up to `max_values` strings into an internal buffer
+    // and reference them into `buffer`.
+    max_values = std::min(max_values, _num_valid_values);
+    if (max_values == 0) {
+        *out_num_values = 0;
+        return Status::OK();
+    }
+
+    int32_t data_size = 0;
+    const int32_t* length_ptr =
+            reinterpret_cast<const int32_t*>(_buffered_length.data()) + _length_idx;
+    for (int i = 0; i < max_values; ++i) {
+        int32_t len = length_ptr[i];
+        if (PREDICT_FALSE(len < 0)) {
+            return Status::InvalidArgument("Negative string delta length");
+        }
+        buffer[i].size = len;
+        if (common::add_overflow(data_size, len, data_size)) {
+            return Status::InvalidArgument("Excess expansion in DELTA_(LENGTH_)BYTE_ARRAY");
+        }
+    }
+    _length_idx += max_values;
+
+    _buffered_data.resize(data_size);
+    if (!_bit_reader->UnpackBatch(8, data_size, _buffered_data.data())) {
+        return Status::IOError("Get length bytes EOF");
+    }
+    uint8_t* data_ptr = _buffered_data.data();
+
+    for (int i = 0; i < max_values; ++i) {
+        buffer[i].data = reinterpret_cast<char*>(data_ptr);
+        data_ptr += buffer[i].size;
+    }
+    // this->num_values_ -= max_values;
+    _num_valid_values -= max_values;
+    *out_num_values = max_values;
+    return Status::OK();
+}
+
+Status DeltaByteArrayDecoder::_get_internal(Slice* buffer, int max_values, int* out_num_values) {
+    // Decode up to `max_values` strings into an internal buffer
+    // and reference them into `buffer`.
+    max_values = std::min(max_values, _num_valid_values);
+    if (max_values == 0) {
+        *out_num_values = max_values;
+        return Status::OK();
+    }
+
+    int suffix_read;
+    RETURN_IF_ERROR(_suffix_decoder.decode(buffer, max_values, &suffix_read));
+    if (PREDICT_FALSE(suffix_read != max_values)) {
+        return Status::EndOfFile("Read " + std::to_string(suffix_read) + ", expecting " +
```

Comment:
please remove all `EndOfFile`, and use `Error("{}, {}, {}", str1, str2, str3)` style.